model/command_autocomplete.go

```diff
@@ -0,0 +1,398 @@
+		return errors.Wrapf(err, "Can't unmarshal argument %s", string(b))
+	}
+	name, ok := arg["Name"]
+	if !ok {
+		return errors.Errorf("No field Name in the argument %s", string(b))
+	}
+	a.Name = name.(string)
+
+	helpText, ok := arg["HelpText"]
+	if !ok {
+		return errors.Errorf("No field HelpText in the argument %s", string(b))
+	}
+	a.HelpText = helpText.(string)
+
+	argType, ok := arg["Type"]
+	if !ok {
+		return errors.Errorf("No field Type in the argument %s", string(b))
+	}
+	a.Type = argType.(string)
+
+	data, ok := arg["Data"]
+	if !ok {
+		return errors.Errorf("No field Data in the argument %s", string(b))
+	}
+
+	if a.Type == AutocompleteTextArgType {
+		m := data.(map[string]interface{})
+		pattern, ok := m["Pattern"]
+		if !ok {
+			return errors.Errorf("No field Pattern in the TextInput argument %s", string(b))
+		}
+		hint, ok := m["Hint"]
+		if !ok {
+			return errors.Errorf("No field Hint in the TextInput argument %s", string(b))
+		}
+		a.Data = &AutocompleteTextArg{Hint: hint.(string), Pattern: pattern.(string)}
+	} else if a.Type == AutocompleteStaticListArgType {
+		m := data.(map[string]interface{})
+		listInterface, ok := m["PossibleArguments"]
+		if !ok {
+			return errors.Errorf("No field PossibleArguments in the StaticList argument %s", string(b))
+		}
+
+		list := listInterface.([]interface{})
+		possibleArguments := []AutocompleteStaticListIem{}
+		for i := range list {
+			args := list[i].(map[string]interface{})
+			itemInt, ok := args["Item"]
+			if !ok {
+				return errors.Errorf("No field Item in the StaticList's possible arguments %s", string(b))
+			}
+			item := itemInt.(string)
+
+			helpTextInt, ok := args["HelpText"]
+			if !ok {
+				return errors.Errorf("No field HelpText in the StaticList's possible arguments %s", string(b))
+			}
+			helpText := helpTextInt.(string)
+
+			possibleArguments = append(possibleArguments, AutocompleteStaticListIem{
+				Item:     item,
+				HelpText: helpText,
+			})
+		}
+		a.Data = &AutocompleteStaticListArg{PossibleArguments: possibleArguments}
+	} else if a.Type == AutocompleteDynamicListArgType {
+		m := data.(map[string]interface{})
+		url, ok := m["FetchURL"]
+		if !ok {
+			return errors.Errorf("No field FetchURL in the DynamicList's argument %s", string(b))
+		}
+		a.Data = &AutocompleteDynamicListArg{FetchURL: url.(string)}
+	}
+	return nil
+}
+
+// NewAutocompleteStaticListArg returned empty AutocompleteStaticListArgType argument.
+func NewAutocompleteStaticListArg() *AutocompleteStaticListArg {
+	return &AutocompleteStaticListArg{
+		PossibleArguments: []AutocompleteStaticListIem{},
+	}
+}
+
+// AddArgument adds a static argument to the StaticList argument.
+func (a *AutocompleteStaticListArg) AddArgument(text, helpText string) {
+	argument := AutocompleteStaticListIem{
+		Item:     text,
+		HelpText: helpText,
+	}
+	a.PossibleArguments = append(a.PossibleArguments, argument)
+}
+
+func stringNotInSlice(a string, slice []string) bool {
```

Comment:
I would imagine that this function is used/exported elsewhere in the code. Not a huge deal if need to define it here.